### ISSUE TITLE
Cannot run Peons because of port conflict

### ISSUE THREAD

**Issue by maver1ck** [apache/incubator-druid#6626]
I'm trying to run KIS on Druid 0.12.3.
And I'm getting following exception, I tried to change both druid.plaintextPort and druid.indexer.runner.startPort but without success. 
Everytime Peon is using druid.plaintextPort. Any idea why ?
```
2018-11-14T13:45:47,846 INFO [task-runner-0-priority-0] io.druid.curator.discovery.CuratorDruidNodeAnnouncer - Announcing [DiscoveryDruidNode{druidNode=DruidNode{serviceName='druid/middleManager', host='100.96.4.69', port=-1, plaintextPort=8092, enablePlaintextPort=true, tlsPort=-1, enableTlsPort=false}, nodeType='peon', services={dataNodeService=DataNodeService{tier='_default_tier', maxSize=0, type=indexer-executor, priority=0}, lookupNodeService=LookupNodeService{lookupTier='__default'}}}].
2018-11-14T13:45:47,910 INFO [task-runner-0-priority-0] io.druid.curator.discovery.CuratorDruidNodeAnnouncer - Announced [DiscoveryDruidNode{druidNode=DruidNode{serviceName='druid/middleManager', host='100.96.4.69', port=-1, plaintextPort=8092, enablePlaintextPort=true, tlsPort=-1, enableTlsPort=false}, nodeType='peon', services={dataNodeService=DataNodeService{tier='_default_tier', maxSize=0, type=indexer-executor, priority=0}, lookupNodeService=LookupNodeService{lookupTier='__default'}}}].
2018-11-14T13:45:47,932 INFO [task-runner-0-priority-0] io.druid.segment.realtime.appenderator.AppenderatorImpl - Created Appenderator for dataSource[platform_game_transactions].
2018-11-14T13:45:48,013 INFO [task-runner-0-priority-0] io.druid.segment.realtime.appenderator.AppenderatorImpl - Loading sinks from[var/druid/task/index_kafka_platform_game_transactions_78e4fbda3095afa_manghdeh/work/persist]: []
2018-11-14T13:45:48,015 INFO [task-runner-0-priority-0] io.druid.segment.realtime.appenderator.StreamAppenderatorDriver - Restored metadata[null].
2018-11-14T13:45:48,018 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding io.druid.server.initialization.jetty.CustomExceptionMapper to GuiceManagedComponentProvider with the scope "Singleton"
2018-11-14T13:45:48,020 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding io.druid.server.initialization.jetty.ForbiddenExceptionMapper to GuiceManagedComponentProvider with the scope "Singleton"
2018-11-14T13:45:48,021 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider to GuiceManagedComponentProvider with the scope "Singleton"
2018-11-14T13:45:48,023 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding com.fasterxml.jackson.jaxrs.smile.JacksonSmileProvider to GuiceManagedComponentProvider with the scope "Singleton"
2018-11-14T13:45:48,036 INFO [task-runner-0-priority-0] io.druid.indexing.kafka.KafkaIndexTask - Seeking partition[0] to offset[0].
2018-11-14T13:45:49,031 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding io.druid.server.http.security.StateResourceFilter to GuiceInstantiatedComponentProvider
2018-11-14T13:45:49,145 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding io.druid.server.http.SegmentListerResource to GuiceManagedComponentProvider with the scope "PerRequest"
2018-11-14T13:45:49,203 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding io.druid.server.QueryResource to GuiceInstantiatedComponentProvider
2018-11-14T13:45:49,209 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding io.druid.segment.realtime.firehose.ChatHandlerResource to GuiceInstantiatedComponentProvider
2018-11-14T13:45:49,214 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding io.druid.server.http.security.ConfigResourceFilter to GuiceInstantiatedComponentProvider
2018-11-14T13:45:49,221 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding io.druid.query.lookup.LookupListeningResource to GuiceInstantiatedComponentProvider
2018-11-14T13:45:49,240 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding io.druid.query.lookup.LookupIntrospectionResource to GuiceInstantiatedComponentProvider
2018-11-14T13:45:49,242 INFO [main] com.sun.jersey.guice.spi.container.GuiceComponentProviderFactory - Binding io.druid.server.StatusResource to GuiceManagedComponentProvider with the scope "Undefined"
2018-11-14T13:45:49,308 WARN [main] com.sun.jersey.spi.inject.Errors - The following warnings have been detected with resource and/or provider classes:
  WARNING: A HTTP GET method, public void io.druid.server.http.SegmentListerResource.getSegments(long,long,long,javax.servlet.http.HttpServletRequest) throws java.io.IOException, MUST return a non-void type.
2018-11-14T13:45:49,319 INFO [main] org.eclipse.jetty.server.handler.ContextHandler - Started o.e.j.s.ServletContextHandler@71f0806b{/,null,AVAILABLE}
2018-11-14T13:45:49,321 ERROR [main] io.druid.cli.CliPeon - Error when starting up.  Failing.
java.net.BindException: Address already in use
        at sun.nio.ch.Net.bind0(Native Method) ~[?:1.8.0_192]
        at sun.nio.ch.Net.bind(Net.java:433) ~[?:1.8.0_192]
        at sun.nio.ch.Net.bind(Net.java:425) ~[?:1.8.0_192]
        at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:223) ~[?:1.8.0_192]
        at sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:74) ~[?:1.8.0_192]
        at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:317) ~[jetty-server-9.3.19.v20170502.jar:9.3.19.v20170502]
        at org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:80) ~[jetty-server-9.3.19.v20170502.jar:9.3.19.v20170502]
        at org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:235) ~[jetty-server-9.3.19.v20170502.jar:9.3.19.v20170502]
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68) ~[jetty-util-9.3.19.v20170502.jar:9.3.19.v20170502]
        at org.eclipse.jetty.server.Server.doStart(Server.java:401) ~[jetty-server-9.3.19.v20170502.jar:9.3.19.v20170502]
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68) ~[jetty-util-9.3.19.v20170502.jar:9.3.19.v20170502]
        at io.druid.server.initialization.jetty.JettyServerModule$1.start(JettyServerModule.java:315) ~[druid-server-0.12.3.jar:0.12.3]
        at io.druid.java.util.common.lifecycle.Lifecycle.start(Lifecycle.java:311) ~[java-util-0.12.3.jar:0.12.3]
        at io.druid.guice.LifecycleModule$2.start(LifecycleModule.java:134) ~[druid-api-0.12.3.jar:0.12.3]
        at io.druid.cli.GuiceRunnable.initLifecycle(GuiceRunnable.java:101) [druid-services-0.12.3.jar:0.12.3]
        at io.druid.cli.CliPeon.run(CliPeon.java:301) [druid-services-0.12.3.jar:0.12.3]
        at io.druid.cli.Main.main(Main.java:116) [druid-services-0.12.3.jar:0.12.3]
^C
```

**Comment by QiuMM** [comment]
It's a bug, you can check #6202 for more details.

**Comment by maver1ck** [comment]
I found a workaround. 
Change plaintextPort to port in middleManager config.

**Comment by jihoonson** [comment]
Closing this issue since the bug was already fixed in #6202.
